File: hmvvm/src/main/java/com/see99/hmvvm/dagger/component/ActivityComponent.java
package com.see99.hmvvm.dagger.component;

import com.see99.hmvvm.base.BaseActivity;
import com.see99.hmvvm.dagger.module.ActivityModule;
import com.see99.hmvvm.dagger.scope.ActivityScope;
import com.see99.hmvvm.ui.login.LoginActivity;

import dagger.BindsInstance;
import dagger.Component;

@ActivityScope
@Component(modules = {ActivityModule.class}, dependencies = {AppComponent.class})
public interface ActivityComponent {


//    void inject(MainActivity activity);

    void inject(LoginActivity loginActivity);

//    @Component.Builder
//    interface Builderx{
//        @BindsInstance
//        Builderx baseAcitvity(BaseActivity<?,?> activity);
//        ActivityComponent build();
//    }

    @Component.Builder
    interface Builder{

        @BindsInstance
        Builder activity(BaseActivity<?,?> activity);

        Builder setAppComponent(AppComponent appComponent);
        ActivityComponent build();
    }
}

File: hmvvmkotlin/src/main/java/com/see99/hmvvmkotlin/dagger/component/ActivityComponent.kt
package com.see99.hmvvmkotlin.dagger.component

import com.see99.hmvvmkotlin.base.BaseActivity
import com.see99.hmvvmkotlin.dagger.module.ActivityModule
import com.see99.hmvvmkotlin.dagger.scope.ActivityScope
import com.see99.hmvvmkotlin.ui.login.LoginActivity
import com.see99.hmvvmkotlin.ui.main.MainActivity
import dagger.BindsInstance
import dagger.Component

@ActivityScope
@Component(modules = [ActivityModule::class], dependencies = [AppComponent::class])
interface  ActivityComponent {
    fun inject(loginActivity: LoginActivity)
    fun inject(loginActivity: MainActivity)

//    fun inject(activity: LoginActivity)

    @Component.Builder
    interface Builder{
        @BindsInstance
        fun activity(activity: BaseActivity<*,*>): Builder

        fun appComponent(component: AppComponent): Builder

        fun build(): ActivityComponent
    }
}